### PR TITLE
fix: code optimization

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | ref #42464 |
| 🇨🇳 Chinese | 无需纳入ChangeLog |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 63423cc</samp>

Fix affix positioning bug and refactor code in `components/affix/index.tsx`.

### <samp>🤖 Generated by Copilot at 63423cc</samp>

* Fix bug where affix component does not update position when target element changes dynamically ([link](https://github.com/ant-design/ant-design/pull/42483/files?diff=unified&w=0#diff-18949a009c0a2f7f47a0136b5808eade7a157b3d78661edc0f75463594e2ad5dL93-R94), [link](https://github.com/ant-design/ant-design/pull/42483/files?diff=unified&w=0#diff-18949a009c0a2f7f47a0136b5808eade7a157b3d78661edc0f75463594e2ad5dL109-R115))
* Refactor and optimize code by merging loops that remove event listeners from target elements ([link](https://github.com/ant-design/ant-design/pull/42483/files?diff=unified&w=0#diff-18949a009c0a2f7f47a0136b5808eade7a157b3d78661edc0f75463594e2ad5dL141-R140))

### CODE REVIEW

File: components/affix/index.tsx [hunk 1]
```diff
@@ -106,15 +106,13 @@ class InternalAffix extends React.Component<InternalAffixProps, AffixState> {
     const targetFunc = this.getTargetFunc();
     const newTarget = targetFunc?.() || null;
     if (prevTarget !== newTarget) {
+      TRIGGER_EVENTS.forEach((eventName) => {
```
Comment (RexSkz) [inline components/affix/index.tsx:109]: 我感觉整个 105~118 行可以抽成一个方法（例如 `this.addListeners`），这个逻辑可以不经修改、直接复用到 `componentDidMount` 的计时器中：

```ts
componentDidMount() {
  this.timer = setTimeout(() => {
    this.addListeners();
  });
}
```

然后把后面的 `componentWillUnmount` 的逻辑（134~143 行）也抽成一个方法（`removeListeners`），跟 `addListeners` 放在相邻的位置，这样会让思路更清晰一些。

File: components/affix/index.tsx [hunk 2]
```diff
@@ -138,15 +108,33 @@ class InternalAffix extends React.Component<InternalAffixProps, AffixState> {
     const newTarget = targetFunc?.();
     TRIGGER_EVENTS.forEach((eventName) => {
       newTarget?.removeEventListener(eventName, this.lazyUpdatePosition);
+      prevTarget?.removeEventListener(eventName, this.lazyUpdatePosition);
```
Comment (MadCcc) [inline components/affix/index.tsx:111]: 这里不等价吧

Comment (RexSkz) [reply components/affix/index.tsx:111]: `prevTarget` 为空的时候这个不会执行，`prevTarget === newTarget` 的时候重复执行也没有问题，所以应该是等价的。

**Comment by afc163** [comment]
感觉应该顺手改成 function component，这个生命周期写起来真蛋疼。

**Comment by li-jia-nan** [comment]
> 感觉应该顺手改成 function component，这个生命周期写起来真蛋疼。
m
是的，是有这个打算，一步一步来，先把恶心的部分重构掉，方便后面改 FC，这是目前除了 rc 之外唯一一个 class 组件了

**Review by afc163 — APPROVED** [review]

**Pull request merged**